packages/react-storage/src/components/StorageBrowser/actions/handlers/utils.ts

```diff
@@ -54,28 +82,68 @@ export const parseLocationAccess = (location: LocationAccess): LocationData => {
     }
   }
 
-  return { bucket, id, permission, prefix, type };
+  let permissions: LocationPermissions;
+  switch (permission) {
+    case 'READ':
+      permissions = ['get', 'list'];
+      break;
+    case 'READWRITE':
+      permissions = ['delete', 'get', 'list', 'write'];
+      break;
+    case 'WRITE':
+      permissions = ['delete', 'write'];
+      break;
+    default:
+      throw new Error(`Invalid location permission: ${permission}`);
+  }
+
+  return { bucket, id, permissions: permissions, prefix, type };
+};
+
+export interface ExcludeType {
+  exactPermissions?: LocationPermissions;
+  type?: LocationType | LocationType[];
+}
```

Comment:
Since permissions are represented in array now, the previous `ExcludeType`(`Permission | LocationType`) can be very confusing, especially in `shouldExcludeLocation`, where we need to handle `exclude: ExcludeType | ExcludeType[]`

The previous interface would complicate the `shouldExcludeLocaton`. For example we need to handle all of the following inputs. 
```
exclude: ['get', 'list']
exclude: [['get', 'list']]
exclude: [['get', 'list'], 'OBJECT'] 
exclude: ['OBJECT', 'BUCKET']
exclude: 'OBJECT'
```

However the following representation is more readable and makes the implementation easier:
```
exclude: { exactPermissions: ['get', 'list'] }
exclude: { exactPermissions: ['get', 'list'], type: 'OBJECT' },
exclude: { type: ['OBJECT', 'BUCKET'] },
```

Reply:
Got it, makes sense. Let's move the declaration to the sibling _types.ts_ file and update the definition to:
```
interface ListLocationsExcludeOptions {
  permissions?: LocationPermissions;
  type?: LocationType | LocationType[]
}
```